rust-bins/src/bin/keygen.rs

```diff
@@ -255,6 +377,55 @@ fn handle_generate_ip_keys(kgip: KeygenIp) -> Result<(), String> {
     Ok(())
 }
 
+fn handle_generate_randomness(grand: GenRand) -> Result<(), String> {
+    // Read word list and make sure it contains 2048 words.
+    let bip39_vec: Vec<_> = include_str!("data/BIP39English.txt")
+        .split_whitespace()
+        .collect();
+    if bip39_vec.len() != 2048 {
+        return Err("The BIP39 word list must contain 2048 words.".to_string());
+    }
+
+    // Create hashmap mapping word to its index in the list.
+    // This allows to quickly test membership and convert words to their index
+    let mut bip39_map = HashMap::new();
+    for (i, word) in bip39_vec.iter().enumerate() {
+        bip39_map.insert(*word, i);
+    }
```

Comment:
I tried to have it in a separate function but got didn't get it to work because of differences between &str and String I don't really understand. Please suggest something that works.